src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -160,7 +160,15 @@ class MarathonSchedulerActor private (
               self ! cmd.answer
               cmd.answer
             }
-            .recover { case ex => CommandFailed(cmd, ex) }
+            .recover {
+              case ex =>
+                logger.error(s"Scale run spec failed. runSpecId=$runSpecId", ex)
+                CommandFailed(cmd, ex)
+            }
+
+          // Always release the lock.
+          result.onComplete(_ => self ! ReleaseLock(runSpecId))
```

Comment:
I was thinking about this, do we want to double release the lock? We may end up releasing a lock that we did not acquire.

If the locks had an id, we could tell the locking system to only release our own lock if it is acquired. That would allow us to have a more fail-proof lock release system.

Barring that, we should probably only release the lock for pathways that we know won't release it :'(

Reply:
I would enable lock release only when `ReleaseLock` is sent.